README_fr.md

```diff
@@ -0,0 +1,1641 @@
+
+```vim
+[alias]
+    a = add
+    amend = commit --amend
+    c = commit
+    ca = commit --amend
+    ci = commit -a
+    co = checkout
+    d = diff
+    dc = diff --changed
+    ds = diff --staged
+    extend = commit --amend -C HEAD
+    f = fetch
+    loll = log --graph --decorate --pretty=oneline --abbrev-commit
+    m = merge
+    one = log --pretty=oneline
+    outstanding = rebase -i @{u}
+    reword = commit --amend --only
+    s = status
+    unpushed = log @{u}
+    wc = whatchanged
+    wip = rebase -i @{u}
+    zap = fetch -p
+```
+
+### Je veux ajouter un répertoire vide à mon dépôt
+
+Vous ne pouvez pas ! Git ne supporte pas cela, mais il y a une astuce. Vous pouvez créer un fichier `.gitignore` dans le dossier avec le contenu suivant :
+
+```
+ # Ignorer tout dans ce dossier
+ *
+ # Sauf ce fichier
+ !.gitignore
+```
+
+Une autre convention commune est de créer un fichier vide dans le dossier, intitulé `.gitkeep` :
+
+```sh
+$ mkdir mondossier
+$ touch mondossier/.gitkeep
+```
+
+Vous pouvez aussi juste appeler le fichier `.keep`, dans ce cas la commande ci-dessus serait `touch mydir/.keep`.
+
+### Je veux mettre en cache un nom d'utilisateur et mot de passe pour un dépôt
+
+Il se peut que vous ayez un dépôt qui nécessite une authentification. Dans ce cas, vous pouvez mettre en cache un nom d'utilisateur et un mot de passe afin que vous n'ayez pas à les rentrer à chaque push / pull. Cette commande peut faire cela pour vous :
+
+```sh
+$ git config --global credential.helper cache
+# Configure Git pour utiliser le cache mémoire des identifiants
+```
+
+```sh
+$ git config --global credential.helper 'cache --timeout=3600'
+# Configurer le cache pour expirer après une heure (le paramètre est en secondes)
+```
+
+### Je veux que Git ignore les changements de permissions et de filemode
+
+```sh
+$ git config core.fileMode false
+```
+
+Si vous voulez configurer cela comme étant le comportement par défaut pour les utilisateurs connectés, utilisez :
+
+```sh
+$ git config --global core.fileMode false
+```
+
+### Je veux définir un utilisateur global
+
+Pour configurer des informations d'utilisateur à travers tous les dépôts locaux, et pour configurer un nom qui est identifiable en relisant l'historique :
+
+```sh
+$ git config --global user.name “[prénom nom]”
+```
+
+Pour configurer une adresse email qui sera associée à chaque ligne de l'historique :
+
+```sh
+git config --global user.email “[adresse-email-valide]”
+```
+
+### Je veux ajouter une ligne de commande de coloration pour Git
+
+Pour configurer la colorisation de ligne de commande automatique de Git afin de faciliter la relecture :
+
+```sh
+$ git config --global color.ui auto
+```
+
+## Je n'ai aucune idée de ce que j'ai mal fait
+
+Donc, vous êtes fichu - vous avez `reset` quelque chose, ou vous avez
```

Comment:
```suggestion
C'est là qu'intervient `git reflog`. `reflog` garde trace de tous les changements du bout de la branche, même si ce dernier n'est pas référencé par une branche ou un tag. Fondamentalement, chaque fois que le HEAD change, une nouvelle entrée est ajoutée au reflog. Cela marche seulement pour les dépôts locaux, malheureusement, et ne trace que les mouvements (pas les changements d'un fichier qui n'ont été enregistrés nulle part, par exemple).
```
It's always "nulle", using the feminine form – it's the **part** that's "nulle".

Reply:
Hm, the "suggested changes" thing is still in Beta, and I think something broke here.
I see the deleted `diff` corresponding to line `1576` while I'm actually editing line `1578`.

Just be careful when applying these changes, I'm not sure what's gonna happen actually.